examples/src/main/java/io/grpc/examples/preserialized/PreSerializedServer.java

```diff
@@ -0,0 +1,159 @@
+        try {
+          PreSerializedServer.this.stop();
+        } catch (InterruptedException e) {
+          e.printStackTrace(System.err);
+        }
+        System.err.println("*** server shut down");
+      }
+    });
+  }
+
+  private void stop() throws InterruptedException {
+    if (server != null) {
+      server.shutdown().awaitTermination(30, TimeUnit.SECONDS);
+    }
+  }
+
+  /**
+   * Await termination on the main thread since the grpc library uses daemon threads.
+   */
+  private void blockUntilShutdown() throws InterruptedException {
+    if (server != null) {
+      server.awaitTermination();
+    }
+  }
+
+  /**
+   * Main launches the server from the command line.
+   */
+  public static void main(String[] args) throws IOException, InterruptedException {
+    final PreSerializedServer server = new PreSerializedServer();
+    server.start();
+    server.blockUntilShutdown();
+  }
+
+  static class GreeterImpl implements GreeterGrpc.AsyncService, BindableService {
+
+    public void byteSayHello(HelloRequest req, StreamObserver<byte[]> responseObserver) {
+      HelloReply reply = HelloReply.newBuilder().setMessage("Hello " + req.getName()).build();
+      responseObserver.onNext(reply.toByteArray());
+      responseObserver.onCompleted();
+    }
+
+    @Override
+    public ServerServiceDefinition bindService() {
+      MethodDescriptor<HelloRequest, HelloReply> sayHello = GreeterGrpc.getSayHelloMethod();
+      // Modifying the method descriptor to use bytes as the response, instead of HelloReply. By
+      // adjusting toBuilder() you can choose which of the request and response are bytes.
+      MethodDescriptor<HelloRequest, byte[]> byteSayHello = sayHello
+          .toBuilder(sayHello.getRequestMarshaller(), new ByteArrayMarshaller())
+          .build();
+      // GreeterGrpc.bindService() will bind every service method, including sayHello(). (Although
+      // Greeter only has one method, this approach would work for any service.) AsyncService
+      // provides a default implementation of sayHello() that returns UNIMPLEMENTED, and that
+      // implementation will be used by bindService(). replaceMethod() will rewrite that method to
+      // use our byte-based method instead.
+      //
+      // The generated bindService() uses ServerCalls to make RPC handlers. Since the generated
+      // bindService() won't expect byte[] in the AsyncService, this uses ServerCalls directly. It
+      // isn't as convenient, but it behaves the same as a normal RPC handler.
+      return replaceMethod(
+          GreeterGrpc.bindService(this),
+          byteSayHello,
+          ServerCalls.asyncUnaryCall(this::byteSayHello));
+    }
+
+    /** Rewrites the ServerServiceDefinition replacing one method's definition. */
+    private static <ReqT, RespT> ServerServiceDefinition replaceMethod(
+        ServerServiceDefinition def,
+        MethodDescriptor<ReqT, RespT> desc,
```

Comment:
For replace with 3 arguments there is an expected pattern of (container, old_value, new_value).  Naming  the arguments with a "new" prefix (desc -> "newDesc" and handler -> "newHandler") would help overcome the tendency to assume the stereotypical pattern.